extensions/amp-subscriptions/0.1/platform-store.js

```diff
@@ -171,18 +235,61 @@ export class EntitlementStore {
   }
 
   /**
-   * Returns most qualified platform
-   * @param {!Array<!./entitlement.Entitlement>} entitlements
-   * @returns {!./entitlement.Entitlement}
+   * Returns most qualified platform.
+   * Qualification of a platform is based on an integer weight.
+   * Every platform starts with weight 0 and evaluated against the following parameters,
+   * - user is subscribed with platform (Gives weight 10)
+   * - supports the current viewer (Gives weight 9)
+   *
+   * In the end candidate with max weight is selected.
+   * However if candidate's weight is equal to local platform, then local platform is selected.
+   * @returns {!./subscription-platform.SubscriptionPlatform}
    */
-  selectApplicablePlatform_(entitlements) {
-    let chosenPlatform;
-    entitlements.forEach(platform => {
-      // TODO(@prateekbh): add metering logic here
-      if (platform.enablesThis()) {
-        chosenPlatform = platform;
+  selectApplicablePlatform_() {
+    const localPlatform = this.getLocalPlatform();
+
+    /** @type {!Object<string, number>} */
+    const platformWeights = {};
+
+    this.getAllResolvedPlatforms().forEach(platform => {
+      const entitlement =
+          this.getResolvedEntitlementFor(platform.getServiceId());
```

Comment:
You are iterating here over getAllResolvedPlatforms(), while calling getResolvedEntitlemntFor. But they are not resolved in tandem (one simply depends on subscriptionplatforms_ var, the other on entitlements_ var). Thus you are spilling too much asynchronous state via synchronous and, it looks to me, bugs are very likely.

Also, we need to minimize public methods returning synchronous/resolved stuff - there shouldn't be any need for them. It's ok to have some minimal API for local platform, but everything else should be async.

To fix this particular issue, however, it looks like you can simply need to ignore a platform that doesn't have entitlements yet. Or you can wait until all platforms have been resolved, but that's not what `getAllResolvedPlatforms` does know as far as I can tell.

Reply:
`getAllResolvedPlatforms()` is just wrongly named and is more like an iteration over the resolved ones. 

Also the entire thing starts after `getAllPlatformsEntitlements_` so it should be safe to assume, we might not run into race condition thingy